src/renderers/WebGLRenderer.js

```diff
@@ -719,31 +719,55 @@ function WebGLRenderer( parameters = {} ) {
 		//
 
 		let index = geometry.index;
-		const position = geometry.attributes.position;
+		let rangeFactor = 1;
+
+		if ( material.wireframe === true ) {
+
+			index = geometries.getWireframeAttribute( geometry );
+			rangeFactor = 2;
+
+		}
 
 		//
 
-		if ( index === null ) {
+		let drawStart = 0;
+		let drawEnd = Infinity;
 
-			if ( position === undefined || position.count === 0 ) return;
+		const drawRange = geometry.drawRange;
+		const position = geometry.attributes.position;
 
-		} else if ( index.count === 0 ) {
+		if ( drawRange !== null ) {
 
-			return;
+			drawStart = Math.max( drawStart, drawRange.start * rangeFactor );
+			drawEnd = Math.min( drawEnd, ( drawRange.start + drawRange.count ) * rangeFactor );
 
 		}
 
-		//
+		if ( group !== null ) {
 
-		let rangeFactor = 1;
+			drawStart = Math.max( drawStart, group.start * rangeFactor );
+			drawEnd = Math.min( drawEnd, ( group.start + group.count ) * rangeFactor );
 
-		if ( material.wireframe === true ) {
+		}
 
-			index = geometries.getWireframeAttribute( geometry );
-			rangeFactor = 2;
+		if ( index !== null ) {
+
+			drawStart = Math.max( drawStart, 0 );
+			drawEnd = Math.min( drawEnd, index.count );
+
+		} else if ( position !== undefined && position !== null ) {
+
+			drawStart = Math.max( drawStart, 0 );
+			drawEnd = Math.min( drawEnd, position.count );
 
 		}
 
+		const drawCount = drawEnd - drawStart;
+
+		if ( ! ( 0 <= drawCount && drawCount < Infinity ) ) return;
```

Comment:
Why not just `if ( drawCount < 0 )`? I think infinite `drawCount` is allowed?

Reply:
I am not sure if it is, since this value is basically passed directly to `gl.drawArrays`/`gl.drawElements`.

https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/drawArrays

Reply:
I just did a quick test, and somewhat to my surprise, passing `Infinity` doesn't seem to cause any issues (aside from nothing being rendered obviously). However, I am still not sure if it is a good idea though. Is it specified somewhere that in infinite count is valid for `gl.drawArrays`/`gl.drawElements`?

Reply:
Basically, my reasoning is that if the `drawCount` is still `Infinity` at this point, it means that it wasn't set explicitly in `drawRange` or `group`, or implicitly by `index` or a `position` attribute. In that case, I would consider the geometry to be empty/incomplete.

Reply:
That reasoning makes sense to me. Still, could you write the check like so:
```js
if ( drawCount < 0 || drawCount === Infinity ) return;
```

Reply:
Sure, I have no strong preference for how the check is written. Your version does look a bit cleaner, so I've adopted it.